### ISSUE TITLE
Form prefill

### ISSUE THREAD

**Issue by RomainQuidet** [buttercup/buttercup-mobile#71]
Hi,
I've installed the latest app from the appStore (1.1.1), but I can't get my password filled from anywhere. My device (6s, iOS 11.2).
If I open Safari on iOS, I don't see any entry for my credentials.
When I go to the app, I see my archive (from DropBox), it's opened. When I go to a credential, if I tap on the "world" icon, I've got a pop up asking if I want to open this website, I tap YES. But nothing happens.

Is there a bug ? How am I supposed to use your app? 
How can I help you? (I'm a freelance developer on iOS and Android).

**Comment by perry-mitchell** [comment]
Hi @RomainQuidet! Thanks for posting. We currently don't have the feature you describe - form prefilling. We'd need integrations built for this, and we haven't had the time. We'd love to have the ability to fill login forms automatically within mobile safari, but are not currently sure how we'd accomplish this.

With regards to your other statement:

> When I go to a credential, if I tap on the "world" icon, I've got a pop up asking if I want to open this website, I tap YES. But nothing happens.

Are you experiencing an issue with opening the website of an entry? If so, what format is the URL?

**Comment by RomainQuidet** [comment]
Hi. Thanks for your reply and great work ! The app seem promising. I understand prefilling needs some work.

I added 3 entries in my archive using Firefox and sync with Dropbox. The firefox plugin is able to log me in for each one.
None of the entries are opening Safari on my iOS device when I tap on the top right icon. When I tap on URL cell or LoginURL cell, it does copy the url and paste works (but it does not help to login). 

Each URL recorded starts as : www.mysecrethost.com/login.php
and each LoginURL starts as http://www.mysecrethost.com/login.php or https:// ... 
I don't get why 2 entries here, and the purpose of each one as they are similar, except the protocol.

Thanks !

**Comment by perry-mitchell** [comment]
Thanks @RomainQuidet!

As the URLs not opening is not related to form prefill I opened a new issue for those: #73

> I don't get why 2 entries here, and the purpose of each one as they are similar, except the protocol.

The 2 URLs are created by the browser extension. the `URL` field is kind of the root-page/domain where you can find the site. The Login URL is created by the extension when it detects the login page. These have uses for future improvements. For instance, the mobile should actually be trying to use the Login URL instead of the regular URL, which I need to check..

**Comment by apat183** [comment]
For prefill in Safari etc. You could try and do it how lastpass do it. When you are on the site, you click the share button in safari and select app in the activity. 

There is a react native lib https://www.npmjs.com/package/react-native-share-extension which supports both android and ios. Which shows you how this might be done. 

This allows the url to be passed to extension but I'm not sure how you then fill the form in safari or browser. Not sure what access the share extension provides to paste values in form. But it looks like to do this, when you create the extension, you include the correct scheme configurations and it should work for passwords and forms. This means it could be also used in apps that support the login with password managers. 
https://stackoverflow.com/questions/38819916/how-to-make-a-compatible-password-extension

**Comment by pleswi** [comment]
Android P will support new API for autofill including password app function :) https://www.androidpolice.com/2018/03/07/android-p-feature-spotlight-new-autofill-apis-improve-compatibility-password-managers/

**Comment by jeroen7s** [comment]
android o alreaddy has an autofill API
though android P will improve on it for better integration with browsers and such

**Comment by se1exin** [comment]
FYI form prefill has been implemented for iOS in #133 (issue #125) and Android is around the corner in #141 (issue #126).

Also using this comment as a back link from #126 as a reminder to close this issue once Android autofill is complete 🙏

**Comment by perry-mitchell** [comment]
I'll leave this issue open for accessibility reasons, but the conversation should continue in #126. As @se1exin said iOS autofill is already in production 😁
